ddl/ddl_worker.go

```diff
@@ -826,39 +826,29 @@ func updateSchemaVersion(t *meta.Meta, job *model.Job) (int64, error) {
 	switch job.Type {
 	case model.ActionTruncateTable:
 		// Truncate table has two table ID, should be handled differently.
-		err = job.DecodeArgs(&diff.TableID)
-		if err != nil {
-			return 0, errors.Trace(err)
-		}
+		diff.TableID = job.CtxVars[0].(int64)
 		diff.OldTableID = job.TableID
 	case model.ActionCreateView:
-		tbInfo := &model.TableInfo{}
-		var orReplace bool
-		var oldTbInfoID int64
-		if err := job.DecodeArgs(tbInfo, &orReplace, &oldTbInfoID); err != nil {
-			return 0, errors.Trace(err)
-		}
+		tbInfo := job.CtxVars[0].(*model.TableInfo)
+		orReplace := job.CtxVars[1].(bool)
+		oldTbInfoID := job.CtxVars[2].(int64)
```

Comment:
I prefer not to modify the original logic if it's not a must.

Reply:
But why not eliminate lots of unmarshal. Whatever, I reverted. And only partitions are modified.